src/main/java/com/google/step/datamanager/DealManagerDatastore.java

```diff
@@ -105,4 +136,66 @@ public Deal updateDeal(Deal deal) {
     searchManager.putDeal(deal, new ArrayList<>());
     return readDeal(deal.id);
   }
+
+  /** Retrieves deals posted by _ followed by user */
+  private List<Deal> getDealsPublishedByFollowedRestaurantsOrUsers(
+      long userId, String fieldName, String filterAttribute) {
+    List<Deal> dealResults = new ArrayList<>();
+    List<Long> idsOfFollowedFieldName = followManager.getFollowedSomething(userId, fieldName);
+    for (Long id : idsOfFollowedFieldName) {
+      Filter propertyFilter = new FilterPredicate(filterAttribute, FilterOperator.EQUAL, id);
+      Query query = new Query("Deal").setFilter(propertyFilter);
+      PreparedQuery pq = datastore.prepare(query);
+      for (Entity entity : pq.asIterable()) {
+        dealResults.add(readDeal(entity.getKey().getId()));
```

Comment:
make a `transformEntityToDeal` function and call it here and everywhere else that might need it